Tests/Marketplace/copy_and_upload_packs.py

```diff
@@ -141,45 +142,93 @@ def upload_core_packs_config(production_bucket: Bucket, build_number: str, extra
-        logging.critical(f"{GCPConfig.CORE_PACK_FILE_NAME} is missing in {build_bucket.name} bucket, exiting...")
-        sys.exit(1)
+    corepacks_files = [GCPConfig.CORE_PACK_FILE_NAME]
+    corepacks_files.extend(GCPConfig.get_core_packs_unlocked_files(marketplace))
+    logging.debug(f"Updating the following corepacks files: {corepacks_files}")
+    for corepacks_file in corepacks_files:
+        build_corepacks_file_path = os.path.join(build_bucket_base_path, corepacks_file)
+        build_corepacks_blob = build_bucket.blob(build_corepacks_file_path)
+
+        if not build_corepacks_blob.exists():
+            logging.critical(f"{corepacks_file} is missing in {build_bucket.name} bucket, exiting...")
+            sys.exit(1)
 
-    temp_corepacks_file_path = os.path.join(extract_destination_path, GCPConfig.CORE_PACK_FILE_NAME)
-    build_corepacks_blob.download_to_filename(temp_corepacks_file_path)
-    corepacks_file = load_json(temp_corepacks_file_path)
+        logging.info(f"Uploading corepacks file {corepacks_file} to bucket.")
+
+        if corepacks_file == GCPConfig.CORE_PACK_FILE_NAME:
+            # change the storage paths to the prod bucket
+            temp_corepacks_file_path = os.path.join(extract_destination_path, GCPConfig.CORE_PACK_FILE_NAME)
+            build_corepacks_blob.download_to_filename(temp_corepacks_file_path)
+            corepacks_file_content = load_json(temp_corepacks_file_path)
+
+            corepacks_list = corepacks_file_content.get('corePacks', [])
+            try:
+                bucket_corepacks_list = [os.path.join(GCPConfig.GCS_PUBLIC_URL, production_bucket.name, storage_base_path,
+                                                      LATEST_ZIP_REGEX.findall(corepack_path)[0]) for corepack_path in
+                                         corepacks_list]
+            except IndexError:
+                corepacks_list_str = '\n'.join(corepacks_list)
+                logging.exception(f"GCS paths in build bucket corepacks.json file are not of format: "
+                                  f"{GCPConfig.GCS_PUBLIC_URL}/<BUCKET_NAME>/.../content/packs/...\n"
+                                  f"List of build bucket corepacks paths:\n{corepacks_list_str}")
+                sys.exit(1)
+
+            # construct core pack data with public gcs urls
+            core_packs_data = {
+                'corePacks': bucket_corepacks_list,
+                'upgradeCorePacks': corepacks_file_content.get('upgradeCorePacks', []),
+                'buildNumber': build_number,
+            }
+
+            # upload core pack json file to gcs
+            prod_corepacks_file_path = os.path.join(storage_base_path, GCPConfig.CORE_PACK_FILE_NAME)
+            prod_corepacks_blob = production_bucket.blob(prod_corepacks_file_path)
+            prod_corepacks_blob.upload_from_string(json.dumps(core_packs_data, indent=4))
+            logging.success(f"Finished uploading {GCPConfig.CORE_PACK_FILE_NAME} to bucket.")
```

Comment:
Can be "merged" with the same `logging.success` below.
Put it after the if-else (if we failed we've `exit(1)` inside the `if` and the `else`.)
```suggestion
```